src/VsExtension/VsEvents/OnBuildPackageRestorer.cs

```diff
@@ -125,6 +133,13 @@ private void SolutionEvents_AfterClosing()
 
         private void BuildEvents_OnBuildBegin(vsBuildScope scope, vsBuildAction Action)
         {
+            // Reset flags for the final status messsage
+            _hasErrors = false;
+            _canceled = false;
+            _hasMissingPackages = false;
+            _displayRestoreSummary = false;
+            _hasOptOutBeenShown = false;
+
```

Comment:
As you are consolidating variables common to both the restores, you should move _msBuildOutputVerbosity = GetMSBuildOutputVerbositySetting(_dte) to somewhere in this method as well. Currently, it is called from both the methods